lib/index.js

```diff
@@ -193,11 +200,11 @@ Socket.prototype._startPollSend = function () {
 }
 
 Socket.prototype._startPollReceive = function () {
-    if (!this._pollReceive) {
-        this._pollReceive = nn.PollReceiveSocket(this.binding, function (events) {
-            if (events) this._receive();
-        }.bind(this));
-    }
+  if (!this._pollReceive) {
+    this._pollReceive = nn.PollReceiveSocket(this.binding, function () {
+      this.push(this._receive());
+    }.bind(this));
```

Comment:
or here, rather, https://github.com/nickdesaulniers/node-nanomsg/blob/master/src/node_nanomsg.cc#L281 if you're looking at nn_recv, same thing, nn_send seems pretty isomorphic.